types/evidence.go

```diff
@@ -554,6 +555,76 @@ func LightClientAttackEvidenceFromProto(lpb *tmproto.LightClientAttackEvidence)
 // EvidenceList is a list of Evidence. Evidences is not a word.
 type EvidenceList []Evidence
 
+// StringIndented returns a string representation of the evidence.
+func (data *EvidenceList) StringIndented(indent string) string {
+	if data == nil {
+		return "nil-Evidence"
+	}
+	evStrings := make([]string, tmmath.MinInt(len(*data), 21))
+	for i, ev := range *data {
+		if i == 20 {
+			evStrings[i] = fmt.Sprintf("... (%v total)", len(*data))
+			break
+		}
+		evStrings[i] = fmt.Sprintf("Evidence:%v", ev)
+	}
+	return fmt.Sprintf(`EvidenceList{
+%s  %v
+%s}#%v`,
+		indent, strings.Join(evStrings, "\n"+indent+"  "),
+		indent, data.Hash())
+}
+
+// ByteSize returns the total byte size of all the evidence
+func (data *EvidenceList) ByteSize() int64 {
+	if len(*data) != 0 {
+		pb, err := data.ToProto()
+		if err != nil {
+			panic(err)
+		}
+		return int64(pb.Size())
+	}
+	return 0
+}
+
+// FromProto sets a protobuf EvidenceList to the given pointer.
+func (data *EvidenceList) FromProto(eviData *tmproto.EvidenceList) error {
+	if eviData == nil {
+		return errors.New("nil evidence list")
+	}
+
+	eviBzs := make(EvidenceList, len(eviData.Evidence))
+	for i := range eviData.Evidence {
+		evi, err := EvidenceFromProto(&eviData.Evidence[i])
+		if err != nil {
+			return err
+		}
+		eviBzs[i] = evi
+	}
+	*data = eviBzs
+	return nil
+}
+
+// ToProto converts EvidenceList to protobuf
+func (data *EvidenceList) ToProto() (*tmproto.EvidenceList, error) {
+	if data == nil {
+		return nil, errors.New("nil evidence list")
+	}
+
+	evi := new(tmproto.EvidenceList)
```

Comment:
- typically the `new()` method is used much more rarely than you'd expect, more typically you either declare the variable and not initalize it (as in `var evi tmproto.EvidenceList`) or you use the initialization form `evi := &tmproto.EvidenceList{}`
- I wouldn't declare this until you're using it, so replace lines 623 with: 
```go
return &tmproto.EvidenceList{Evidence: eviBzs}, nil` 
```

The reason for the change is to avoid constructing objects before you need (as in the case of the early return,)

As a second comment, initializing slices with lengths is (generally) a good practice, but it's a bit of a premature optimization, so just for some background:
- if you initialize an empty slice (no capacity or length) and then append items to it, the runtime has to reallocate memory in the underlying data structures as you append. This means that for longer lists of things, or in memory pressure situations with fast loops that build lists of data, you can actually observe a performance penalty for `append()`
- there are two solutions, the one you've chosen (make a slice of n zero'd elements, and then iterate over the input slice and set the values of each slice index to the new value that you construct.conversely, you can create a slice with a _capacity_ that has a length of zero (e.g. `make(tmproto.EvidenceList, 0, <n>)`, which means that you can use `append()` as normal, but you've given a hint to the runtime that says "make me a slice that won't need to be resized if  I add `<n`> items to it. While there are definitely cases for the length case, often the capacity case is more clear.